README.md

```diff
@@ -1,121 +1,183 @@
+    - [Command 0x83 = Get bootloader version (master read)](#command-0x83--get-bootloader-version-master-read)
+    - [Command 0x84 = Rewind target address (master write)](#command-0x84--rewind-target-address-master-write)
+    - [Command 0x85 = Read flash memory](#command-0x85--read-flash-memory)
 
-The bootloader makes it possible to update the SMC firmware from the Commander X16 without an external programmer.
+# Project Description
 
+This is a custom bootloader for the Commander X16 ATtiny861 based System Management Controller (SMC).
 
-# How It Works
+The bootloader makes it possible to update the SMC firmware from the Commander X16 without using an external programmer.
 
-## Firmware Update Process
+The firmware is stored at the beginning of the flash memory and provides an interface to the PS/2 keyboard
+and mouse. It also handles the psysical push buttons and the LEDs of the system.
 
-The update process is started either by requesting the firmware to call
-the bootloader's start update function or by holding down the Reset button
-while connecting the computer to power. The latter is called a recovery
-update. Read more about that below.
+The bootloader is a separate program that is stored at the end of the flash memory.
 
-A firmware update program runs on the X16 and sends the new SMC firmware to
-the bootloader over I2C.
 
-The bootloader is responsible for checking the integrity of the transmitted data 
-and for writing it to the flash memory of the SMC.
+# How the Bootloader Works
 
-The transmission is divided into packets of nine bytes. A packet begins with
-eight firmware bytes. The last byte is a checksum, the 
-2's complement of the sum of the previous eight bytes in the packet. I2C command 
-offset 0x80 is used to transmit each byte of a packet.
+## Introduction
 
-A complete packet must committed using I2C command offset 0x81 before transmission
-of the next packet is started.
+For detailed information on how to update the SMC firmware, go to the [SMC Update Guide](https://github.com/X16Community/x16-smc/blob/main/doc/update-guide.md).
 
-The update process continues by transmitting and committing packets until the whole
-firmware has been received by the bootloader.
+This section describes the inner workings of the bootloader. The information is of most interest to those who write tools that use the bootloader.
 
-The update program running on the X16 can optionally rewind the target address
-to 0x0000 using command 0x84 and use command 0x85 to read one byte at a time from
-the SMC flash memory. This can be used to verify the update.
+## Bootloader Entry Points
 
-Finally, the update program must use command 0x82 to reset the SMC. This will
-write any remaining buffered data to flash memory and then turn off the computer.
+The bootloader has two entry points: the main entry point at address 0x1e00, and the start update entry point at address 0x1e02.
```

Comment:
Before jumping straight to the "entry points", can I suggest adding a sentence explaining that all offsets in this document are byte offsets, and quickly mention the memory map (main program 0x0000 to 0x1DFF, bootloader 0x1E00-0x1FFF)?

Reply:
Reading through the entire document, I suggest to explain that all offsets are byte offsets, and add a link to the memory map. I suggest to write this in the introduction section, before the "bootloader entry points" section.